In [2]:
import numpy as np
def fred(a,b):
    c=(a+b)*12
    return c
a = 13.2
b = 3.9
c = fred(a,b)

print("c = ",c)

c =  205.2


In [7]:
#Lets write rule 1
def rule1(c, dA):
    dQ = np.abs(c)*dA
    return dQ
dA = 0.1
dB = 1.1

errT = rule1(c, dA)
print (errT)

20.52


In [10]:
#Lets write rule 2
def rule2(c, m, dA, A):
    dQ = np.abs(c*m*A**(m-1))*dA
    return dQ
A = 1
m = 3
dA = 2
c = 1

errT = rule2(c, m, dA, A)
print (errT)

6


In [11]:
#lets write rule 3
def rule3(dA, dB):
    dQ = np.sqrt(dA**2 + dB**2)
    return dQ
dA = 0.1
dB = 1.1

errT = rule3(dA, dB)
print (errT)


1.104536101718726


In [20]:
#Lets write rule 4
def rule4(Q, m, n, dA, A, dB, B):
    dQ = np.sqrt((m*dA/A)**2 + (n*dB/B)**2)*Q
    return dQ
Q = 0.783
dA = 0.0001
dB = .27
A = 0.2083
B = 10.5
m = 1
n = -1

errT = rule4(dA, dB, Q, m, n, A, B)
print (errT)

3.1149336908708744e-06


In [21]:
# calculate the average, and err in mean
x = np.array([1.1,1.3,1.4,0.9,0.95,1.05])
print(np.std(x))
print(np.average(x))

0.17950549357115014
1.1166666666666667


# the error in average: 0.1795

# the average is 1.12

$\delta Q = \sqrt{(\delta A)^2 + (\delta B)^2}$

$\Sigma F_c =0 (4\pi^2*m)/(slope)$